### PR TITLE
Refactor and update the transformation mapping specs

### PR DESCRIPTION
**Pull request by djberg96**
The specs for transformation mappings need to be updated. The main things are:

* The lans need to be valid, i.e. actually attached to a switch and associated with an EMS.
* We should use factory lists where appropriate instead of creating new, identical factories.
* The source and destination hosts need to be corrected (the redhat and vmware hosts are flip-flopped).

~~WIP until I can get back to this or someone (@thearifismail or @agrare) gets to it first.~~ Guess I was done first.

### CODE REVIEW

File: spec/requests/transformation_mappings_spec.rb [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
-
-  let(:destination_vmware_host) { FactoryBot.create(:host_vmware, :ems_cluster => destination_cluster) }
-  let(:destination_storage) { FactoryBot.create(:storage, :hosts => [destination_vmware_host]) }
-
-  let(:source_lan) { FactoryBot.create(:lan) }
-  let(:destination_lan) { FactoryBot.create(:lan) }
+  let(:source_clusters) { FactoryBot.create_list(:ems_cluster, 2, :ext_management_system => source_ems) }
+  let(:source_hosts) { FactoryBot.create_list(:host_vmware, 1, :ems_cluster => source_clusters.first) }
+  let(:source_storages) { FactoryBot.create_list(:storage, 2, :hosts => source_hosts) }
+  let(:source_switches) { FactoryBot.create_list(:switch, 1, :host => source_hosts.first) }
```
Comment (agrare) [inline spec/requests/transformation_mappings_spec.rb]: This should be `FactoryBot.create_list(:switch, 1, :hosts => source_hosts)`

File: spec/requests/transformation_mappings_spec.rb [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+  let(:source_lans) { FactoryBot.create_list(:lan, 2, :switch => source_switches.first) }
+
+  let(:destination_ems) { FactoryBot.create(:ems_redhat) }
+  let(:destination_clusters) { FactoryBot.create_list(:ems_cluster, 2, :ext_management_system => destination_ems) }
+  let(:destination_hosts) { FactoryBot.create_list(:host_redhat, 1, :ems_cluster => destination_clusters.first) }
+  let(:destination_storages) { FactoryBot.create_list(:storage, 2, :hosts => destination_hosts) }
+  let(:destination_switches) { FactoryBot.create_list(:switch, 1, :host => destination_hosts.first) }
```
Comment (agrare) [inline spec/requests/transformation_mappings_spec.rb]: Same, `FactoryBot.create_list(:switch, 1, :hosts => destination_hosts)`

File: spec/requests/transformation_mappings_spec.rb (hunk 1 above)
Comment (djberg96) [reply spec/requests/transformation_mappings_spec.rb]: Updated.

File: spec/requests/transformation_mappings_spec.rb (hunk 2 above)
Comment (djberg96) [reply spec/requests/transformation_mappings_spec.rb]: Updated, thanks.

**Comment by agrare** [comment]
@miq-bot assign @abellotti

**Comment by agrare** [comment]
@miq-bot add-label test

**Comment by abellotti** [comment]
This looks good and nice use of the lists. Thanks @djberg96 for the updated tests. LGTM!! 👍

**Comment by lpichler** [comment]
this also fixes CI failure https://travis-ci.org/ManageIQ/manageiq-api/jobs/582667184

thanks @djberg96 !

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 14f3cd60f4dd9d817b01d687a13bbb62db28f2a2
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Mon Sep 9 10:11:24 2019 -0400

    Merge pull request #669 from djberg96/transformation_mapping_spec_refactor
    
    Refactor and update the transformation mapping specs
    
    (cherry picked from commit 1add1df801a54815f64254d13e0bd0772220f87d)
```

**Pull request merged**